### ISSUE TITLE
Is it possible to use it with Windows 10's OpenSSH server?

### ISSUE THREAD

**Issue by darkvertex** [sshuttle/sshuttle#496]
In Windows 10 you can enable an optional (official) OpenSSH server component from Microsoft:

https://docs.microsoft.com/en-us/windows-server/administration/openssh/openssh_install_firstuse#:~:text=OpenSSH%20client%20and%20server%20are,OpenSSH%20client%20is%20already%20installed.

I tried to use sshuttle from my Ubuntu machine to a Windows machine to forward 10.0.0.0/8 to no avail so I guess it's unsupported:
```
$ sshuttle -vNHr user@192.168.55.183 10.0.0.0/8
Starting sshuttle proxy.
firewall manager: Starting firewall with Python version 3.6.9
firewall manager: ready method name nat.
IPv6 enabled: False
UDP enabled: False
DNS enabled: False
TCP redirector listening on ('127.0.0.1', 12300).
Starting client with Python version 3.6.9
c : connecting to server...
The system cannot find the path specified.
'P' is not recognized as an internal or external command,
operable program or batch file.
c : fatal: server died with error code 1
```

Any tricks I could use to make it work, like perhaps running a Docker container with an ssh daemon? Would that work?

**Comment by midthegap** [comment]
I'm trying to do the same. I share my findings, hoping for some help. 

If you ask for more verbosity (sudo sshuttle -vvvv ...) you will find that sshuttle tries to execute python on the server in Linux way:
`
['ssh', 'user@192.168.1.220', '--', 'exec /bin/sh -c \'P=python3; $P -V 2>/dev/null || P=python; exec "$P" -c \'"\'"\'import sys, os; verbosity=4; sys.stdin = os.fdopen(0, "rb"); exec(compile(sys.stdin.read(1106), "assembler.py", "exec"))\'"\'"\'\'']
`

I think the error message you get is from Windows, telling you that P is not a valid command.
I tried to add a custom python command:

`sudo sshuttle --dns -r user@192.168.1.220 0.0.0.0/0 -vvvv --exclude 192.168.1.220 --python python.exe`

now Windows tell me that python is not a recognized command. So I tried 

`sudo sshuttle --dns -r client@192.168.1.220 0.0.0.0/0 -vvvv --exclude 192.168.1.220 --python C:\Users\user\AppData\Local\Programs\Python\Python38\python.exe`

But Sshuttle removes slashes, resulting in a not valid command:

`
c : executing: ['ssh', 'user@192.168.1.220', '--', '\'C:UsersuserAppDataLocalProgramsPythonPython38python.exe\' -c \'import sys, os; verbosity=4; sys.stdin = os.fdopen(0, "rb"); exec(compile(sys.stdin.read(1106), "assembler.py", "exec"))\'']
`
Any ideas on how to instruct sshuttle about pyhton on Windows?

**Comment by midthegap** [comment]
Sorry, I gave up with Windows and moved to a Linux distribution.

**Comment by jlatusek** [comment]
Hi, I'm looking for a solution to the same problem. Is there any update?.

**Comment by siouxchief** [comment]
Same here, spent a whole day last week but similar issues to the above

**Comment by yorickreum** [comment]
I'd be interested in doing this too...

**Comment by fishikles** [comment]
I had a play with this, and got it to the point of connecting to Cygwin64's bash terminal via OpenSSH server
However it looks like the ssnet.py file isn't multiplatform, specifically it's trying to set the stdin and stdout to nonblocking which isn't supported under windows as far as I can see.

**Comment by oddrabbit** [comment]
Looks like this issue is eternal.

**Comment by brianmay** [comment]
Fixed by #918.
